### DISCUSSION TITLE
Option templates

### DISCUSSION THREAD

**Discussion by genox** [swellstores/community#190] [Ideas]
Hi,

Currently setting up a store and exploring features. Products are mostly clothing with color and size options.

I was looking for an option to generate product variants based on attributes that can be centrally managed and chosen on a product level. The current way of adding options makes it hard to ensure consistency over multiple products and is rather cumbersome, especially if option strings need to be translated while entering for each product.

One approach would be to create a template product that contains all possible option parameters, like size and colors that are translated and maybe contain additional field data (rgb hex strings).

A user would then remove unneeded options. But then, if he accidentally removed one or a product now become available in another color that was previously removed from the set, external documentation needs to exist that explains what to enter to ensure consistency.

Using an attribute as an option template of sorts and not using a tag field but a multi select would make that much easier.

Did you consider ways to make this process easier?

**Comment by ericingram** [comment]
We've thought about this and I would say it's a worthwhile improvement. It's not very far off because the system already links all variant names and values to an attribute, so there is a common record that could be used for a select-like UI.

I'm sure we'll get to it at some point. If anyone is looking for this, add a +1

**Reply by genox** [reply]
Yeah, I noticed that options are persisted as variants if I saw that right, causing some confusion on my part because suddenly I had two attributes of the same name with one not being editable in the attribute manager.

The way options are set up works great and is fast if the number of products is small and there is only some products with options. For an entire catalogue of cycling apparel it will pose some mid-term content management issues.

Looking forward to this, as I will be a heavy user.

If I start adding products and options now the "normal" way, how difficult will a migration be? I would like to avoid adding everything twice or writing migration code.

**Reply by ericingram** [reply]
Good points. There probably won't be any changes to the data structure, rather the UI for editing and managing options would connect the dots for you.

**Comment by genox** [comment]
Hi,

Can you provide an update on when to expect such an improvement?

**Reply by ericingram** [reply]
It's not prioritized yet as we have a number of more critical changes to make, but would welcome more ideas to help us plan improvements in the future.

**Reply by johnrisby** [reply]
This is something we'd find very helpful. As attributes used to create variants are stored as Attributes anyway, isn't it just a case of changing the Options section so instead of just a free-form field, with no (obvious) connection to existing attributes, it has a searchable-as-you-type select field instead that lets you choose from the existing attributes, along with the option to create a new one? That, at the most basic, would at least prevent mistakes like typing the attribute name incorrectly.

An extension to this though would be limiting the choices for the values to the ones already set in the Attribute, as well as the option for automatically creating variants. That would really fool-proof it.

Is that how you see it @genox ? 

Do you have any idea at all when something like this might be added @ericingram ?

The alternative of course is to create a custom admin UI. We already use PayloadCMS for most of our other content, so it wouldn't be hard to do that but if it could be avoided right now, that'd be great. Thanks

**Reply by genox** [reply]
Swell didn't make the cut for me while evaluating multiple solutions. I settled with a self-hosted Saleor setup.

**Reply by johnrisby** [reply]
Thanks @genox, I've heard that from a few people sadly (not necessarily Saleor btw). Hope your project went/is going well.

**Reply by ericingram** [reply]
Given the flexibility of options I agree this would be a great improvement. We mapped it out a while ago but hasn’t made the cut yet. I am sure we’ll get there though.
